iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -374,9 +373,12 @@ public DynamicPartitionCtx createDPContext(HiveConf hiveConf, org.apache.hadoop.
       fieldOrderMap.put(fields.get(i).name(), i);
     }
 
+    // deletes already use the bucket values in the partition_struct for sorting, so no need to add the sort expression
```

Comment:
Is this true for bucket transform only? Is bucket() special for some reason, or could we avoid sorting according to other partition columns (and transform types) as well?

Reply:
Yes, good catch, I think we can avoid sorting by the other partition columns too.